### PR TITLE
feat: support avatar decorations

### PR DESCRIPTION
**Pull request by sdanialraza**
**Please describe the changes this PR makes and why it should be merged:**

**If applicable, please reference Discord API Docs PRs or commits that influenced this PR:**
- https://github.com/discord/discord-api-docs/pull/6464

### CODE REVIEW

File: payloads/v10/user.ts [hunk 1]
```diff
@@ -84,11 +84,11 @@ export interface APIUser {
...
 	public_flags?: UserFlags;
 	/**
-	 * The user's avatar decoration hash
+	 * The data for the user's avatar decoration
 	 *
-	 * See https://discord.com/developers/docs/reference#image-formatting
+	 * See https://discord.com/developers/docs/resources/user#avatar-decoration-data-object
 	 */
-	avatar_decoration?: string | null;
+	avatar_decoration?: AvatarDecorationData | null;
```
Comment (almeidx) [inline payloads/v10/user.ts:91]: The old field should stay and be marked as deprecated. The new field is called avatar_decoration_data

**Review by almeidx — CHANGES_REQUESTED** [review]

File: deno/payloads/v10/user.ts [hunk 2]
```diff
@@ -84,11 +84,11 @@ export interface APIUser {
...
 	public_flags?: UserFlags;
 	/**
-	 * The user's avatar decoration hash
+	 * The data for the user's avatar decoration
 	 *
-	 * See https://discord.com/developers/docs/reference#image-formatting
+	 * See https://discord.com/developers/docs/resources/user#avatar-decoration-data-object
 	 */
-	avatar_decoration?: string | null;
+	avatar_decoration?: AvatarDecorationData | null;
```
Comment (Jiralite) [inline deno/payloads/v10/user.ts:91]: This is, I assume, now wrong (and a breaking change). `avatar_decoration` should stay as it is but become deprecated. The new property is `avatar_decoration_data`.

File: payloads/v10/user.ts [hunk 3]
```diff
@@ -318,3 +326,19 @@ export interface APIApplicationRoleConnection {
 	 */
 	metadata: Record<string, string | number>;
 }
+
+/**
+ * https://discord.com/developers/docs/resources/user#avatar-decoration-data-object
+ */
+export interface AvatarDecorationData {
```
Comment (almeidx) [inline payloads/v10/user.ts:333]: Missing API prefix

File: payloads/v10/user.ts [hunk 4]
```diff
@@ -318,3 +326,19 @@ export interface APIApplicationRoleConnection {
...
+	/**
+	 * The avatar decoration hash
+	 *
+	 * See https://discord.com/developers/docs/reference#image-formatting
+	 */
+	asset: string;
+	/**
+	 * The id of the avatar decoration's SKU
+	 */
+	sku_id: string;
```
Comment (almeidx) [inline payloads/v10/user.ts:343]: Should be typed as `Snowflake`

**Review by almeidx — CHANGES_REQUESTED** [review]
Can you update the `CDNRoutes` too? (in rest/v10/index.ts)

File: rest/v10/index.ts [hunk 5]
```diff
@@ -1020,12 +1020,12 @@ export const CDNRoutes = {
 
 	/**
 	 * Route for:
-	 * - GET `/avatar-decorations/{user.id}/{user.avatar_decoration}.png`
+	 * - GET `/avatar-decoration-presets/{avatar_decoration_data_asset}.png`
 	 *
 	 * This route supports the extension: PNG
 	 */
-	userAvatarDecoration(userId: Snowflake, userAvatarDecoration: string) {
```
Comment (almeidx) [inline rest/v10/index.ts:1027]: The old function should stay, and be marked as deprecated in favour of the new one.
This change is also missing in rest/v9

**Review by almeidx — CHANGES_REQUESTED** [review]

File: rest/v10/index.ts [hunk 6]
```diff
@@ -1023,11 +1023,23 @@ export const CDNRoutes = {
 	 * - GET `/avatar-decorations/{user.id}/{user.avatar_decoration}.png`
 	 *
 	 * This route supports the extension: PNG
+	 *
+	 * @deprecated Use {@link CDNRoutes.avatarDecoration} instead.
```
Comment (almeidx) [inline rest/v10/index.ts:1027]: This doesn't appear to work with the current documentation https://discord-api-types-git-fork-sdanialraza-feat-su-9a6e74-discordjs.vercel.app/api/next/discord-api-types-v10#CDNRoutes

```suggestion
	 * @deprecated Use {@apilink CDNRoutes#avatarDecoration} instead.
```

**Review by almeidx — CHANGES_REQUESTED** [review]

**Review by almeidx — DISMISSED** [review]

**Review by almeidx — APPROVED** [review]

**Comment by vladfrangu** [comment]
This has been merged 👀

**Review by advaith1 — APPROVED** [review]

File: deno/payloads/v10/user.ts [hunk 7]
```diff
@@ -87,8 +87,16 @@ export interface APIUser {
 	 * The user's avatar decoration hash
 	 *
 	 * See https://discord.com/developers/docs/reference#image-formatting
+	 *
+	 * @deprecated Use `avatar_decoration_data` instead
 	 */
 	avatar_decoration?: string | null;
```
Comment (advaith1) [inline deno/payloads/v10/user.ts:93]: the old avatar_decoration field does not exist anymore (and has not existed for a long time) so we could just delete it, or leave it deprecated

Comment (vladfrangu) [reply deno/payloads/v10/user.ts:93]: Deprecated and always null

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**